### PR TITLE
chore: Added documentation for newly added functionality

### PR DESCRIPTION
**Pull request by abhilash-kumar-nair**
The changes could be viewed on https://modelon-impact-client.readthedocs.io/en/ss-docs-update/. I have made the link temporarily active(Its hidden for normal users)

Please let me know if I missed to capture something in the workflow :)

### CODE REVIEW

File: docs/source/tutorial.rst [hunk 1]
```diff
@@ -313,13 +313,143 @@ set in both ``experiment_definition`` and extensions, the one set in the extensi
...
 ------------------------
 
+Following either of the approaches listed below, the experiment definition set up could be executed:
+
+One step execution
+##################
 The experiment definition set up can now be passed to the ``execute()`` function::
 
-   exp = workspace.execute(experiment_definition).wait()
+   experiment = workspace.execute(experiment_definition).wait()
```
Comment (efredriksson-modelon) [inline docs/source/tutorial.rst:322]: Note that you need to update the code snippet for plotting to use `experiment`.

**Review by efredriksson-modelon — CHANGES_REQUESTED** [review]
Nice work documenting the workflows! 🧇 

I am a little worried that we just have a single tutorial and everything is added to it. It would be good if we could break it up and structure it so the user can take the tutorials they are intressted in. I am happy to help. Here is a suggestion on the outline of the tutorials:
 * Setting up the workspace (includes setting up the client also)
 * Choosing analysis type (just add at the end that this class `CustomFunction` will be used as input for creating an experiment)
 * Choosing a model (just a rename of Set up model for analysis)
 * Setup and run an experiment (Contains "2.1 Setting up an experiment" and "Executing the experiment" and would also have to include the code `experiment_definition = experiment_definition.with_modifiers({'PI.k': Range(10, 100, 3)})` mentioning super short that this will make it so the experiment contains 3 cases and then have a reference to the mutli-execution tutorial for more information on how to do this.)
 * Plotting experiment results (re-name of "Plotting the results")
 * Muti-execution experiments (operators and extensions sections included here)
 * Re-executing and updating experiments (what you have written)
 * FMU based workflow (would also contain the relevant part of 'Analysis' explaining what this is about)
 * Logging output for client (rename of Setting logging levels for analysis)
 * Removing FMUs and experiments

Unrelated but would be very nice to fix. The line
`model = workspace.upload_model_library('C:/Model.mo')`
in documentation is not correct. That method does not return anything so you would still need to create the 'model' class for a specific model yourself.

File: docs/source/tutorial.rst [hunk 2]
```diff
@@ -313,13 +313,143 @@ set in both ``experiment_definition`` and extensions, the one set in the extensi
...
+
+This approach couples both the creation and execution of the experiment in one step. Executing this multiple
+time would create a new experiment object, hence is not advisable if the user wishes to re-execute the same 
+experiment multiple times.
```
Comment (andrewwren-modelon) [inline docs/source/tutorial.rst:326]: ln 325 time should be plural...
Perhaps:
Subsequent executions will create a new experiment object and as such ill advised for re-execution of the same experiment.

File: docs/source/tutorial.rst [hunk 3]
```diff
@@ -313,13 +313,143 @@ set in both ``experiment_definition`` and extensions, the one set in the extensi
...
+
+Two step execution
+##################
+This approach isolate the creation and execution of an experiment as two methods::
```
Comment (andrewwren-modelon) [inline docs/source/tutorial.rst:330]: isolates?

**Comment by abhilash-kumar-nair** [comment]
Thanks for review and the good feedback. @efredriksson-modelon , @andrewwren-modelon . Could you please take another look ? Thanks!

File: docs/source/muti_execution_experiments.rst [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+Operators can be used to create multi-execution experiments. Here is an example where a multi-execution
+experiment definition with three cases is created::
+
+   from modelon.impact.client import Range
+
+   experiment_definition = experiment_definition.with_modifiers({'PI.k': Range(10, 100, 3)})
+
+Here the ``Range()`` operator class is used to specify the ``start_value``, ``end_value`` and ``no_of_steps`` for the parameter to sweep.
+The parametrization in the examples above would configure a set of three simulation cases for the fmu with ``3`` equidistant
```
Comment (efredriksson-modelon) [inline docs/source/muti_execution_experiments.rst:21]: fmu -> model

File: docs/source/reexecuting_updating_experiments.rst [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+   experiment = experiment.execute(with_cases=[case_2, case_3]).wait()
+
+Updating cases
+**************
+
+The case input for a specific case in a batch simulation can be updated and the case can be re-executed.
+The following input values could be updated.
+1. Custom function parameters - E.g. - 'start_time', 'final_time' for a dynamic simulation.
+2. Simulation and solver options - E.g. - 'ncp', 'atol' etc for a dynamic simulation.
+3. Non-structural parameters - E.g. - 'PI.k' for the Modelica.Blocks.Examples.PID_Controller.
```
Comment (efredriksson-modelon) [inline docs/source/reexecuting_updating_experiments.rst:34]: This list is not formatted correctly when I build HTML. Maybe we need an empty line before it?

File: docs/source/logging_output_for_client.rst [hunk 6]
```diff
@@ -0,0 +1,14 @@
+Logging output for client
+=========================
+
+*This tutorial covers setting logging options to control the amount of information output by the client.*
+
+The built-in python module, ``logging`` (https://docs.python.org/3.9/library/logging.html) is used by the client. The amount of logging that
+should be output by the client can be set by specifying the log level as shown in the code snippet.::
```
Comment (efredriksson-modelon) [inline docs/source/logging_output_for_client.rst:7]: .:: -> ::

File: docs/source/logging_output_for_client.rst [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+   import logging
+
+   logging.getLogger('modelon.impact.client').setLevel(logging.ERROR)
+
+The log is printed to the standard output, normally the terminal window from which the compiler is invoked. The available log levels are
```
Comment (efredriksson-modelon) [inline docs/source/logging_output_for_client.rst:13]: compiler?

File: docs/source/reexecuting_updating_experiments.rst [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+
+Filtered case execution
+***********************
+
+It is possible to update and run a subset of cases in a batch simulation. Call the execute method on the 
+experiment object with the ``with_cases`` argument set to a list of the cases that should be executed. 
+Observe that if the experiment has not been executed yet it will not have any cases that we can access. 
+To work around this we can first execute the experiment and specify that no cases should be executed::
+
+    experiment_expand = experiment.execute(with_cases=[]).wait()
```
Comment (efredriksson-modelon) [inline docs/source/reexecuting_updating_experiments.rst:14]: Maybe add the code line 
experiment = workspace.create_experiment(experiment_definition)

to make it more clear that you run this just after creating the experiment.

**Review by efredriksson-modelon — CHANGES_REQUESTED** [review]
Nice I think this should be very helpful for users not intressted in only some parts! 🍉🍉🍉

**Comment by abhilash-kumar-nair** [comment]
> Nice I think this should be very helpful for users not intressted in only some parts! 🍉🍉🍉

Thanks again @efredriksson-modelon for the great feedback! I have made the suggested fixes! Could you please taken another quick look? Thanks!

**Review by efredriksson-modelon — APPROVED** [review]
👍

**Pull request merged**